File: 14-longest-common-prefix/14-longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        int min = Integer.MAX_VALUE;
        for(String str: strs) {
            min = str.length() < min ? str.length() : min;            
        }
        String prefix = "";
        for(char cur=0; cur<min; cur++) {
            char c = strs[0].charAt(cur);
            int i;
            boolean flag = true;
            for(i=0; i<strs.length; i++) {
                if(c == strs[i].charAt(cur)) {
                    continue;
                }
                else {
                    flag = false;
                }
            }
            if(i >= strs.length && flag) {
                prefix += c;
            }
            if(flag == false){
                break;
            }
        }
        return prefix;
    }
}

File: 14-longest-common-prefix/14-longest-common-prefix.js
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function(strs) {
    strs.sort();
    let count = 0;
    for(let i=0; i<Math.min(strs[0].length, strs[strs.length - 1].length); i++) {
        if(strs[0][i] === strs[strs.length - 1][i]) count++;
        else break;
    }
    return strs[0].substring(0, count);
};